### PR TITLE
Add help option to test command line args

### PR DESCRIPTION
**Pull request by Skinner927**
When test binaries are run with unknown options or with the standard -h option, a help menu will print all available options.

This is much more convenient than having to dig through unity.c to find every option.

### CODE REVIEW

**Comment by HaorongX** [comment]
Great idea. I also found it confusing when I have to check `unity.c` to find these options. I think the usage should be added to docs at the same time. ^_^

File: src/unity.c [hunk 1]
```diff
@@ -2329,6 +2329,17 @@ int UnityParseOptions(int argc, char** argv)
                     UnityPrint("ERROR: Unknown Option ");
                     UNITY_OUTPUT_CHAR(argv[i][1]);
                     UNITY_PRINT_EOL();
+                    /* fall-through to display help */
+                case 'h':
+                    UnityPrint("Options: "); UNITY_PRINT_EOL();
+                    UnityPrint("-l        List all tests"); UNITY_PRINT_EOL();
+                    UnityPrint("-f TEST   Only run tests with TEST in the name"); UNITY_PRINT_EOL();
```
Comment (mvandervoord) [inline src/unity.c:2336]: Maybe change the wording to make it obvious what the function letter is? Something like "Filter to run only tests with TEST in the name"?

File: src/unity.c [hunk 2]
```diff
@@ -2329,6 +2329,17 @@ int UnityParseOptions(int argc, char** argv)
...
+                    UnityPrint("-n TEST   Only run tests with TEST in the name"); UNITY_PRINT_EOL();
```
Comment (mvandervoord) [inline src/unity.c:2337]: I'd not include the `n` option... or if I did include it, mark it as `deprecated`?

File: test/tests/test_generate_test_runner.rb [hunk 3]
```diff
@@ -1158,9 +1158,43 @@
       :to_pass => [ ],
       :to_fail => [ ],
       :to_ignore => [ ],
-      :text => [ "ERROR: Unknown Option z" ],
+      :text => [
+        "ERROR: Unknown Option z",
+        "Options:",
+        "-l        List all tests",
+        "-f TEST   Only run tests with TEST in the name",
+        "-n TEST   Only run tests with TEST in the name",
```
Comment (mvandervoord) [inline test/tests/test_generate_test_runner.rb:1166]: see above

**Review by mvandervoord — CHANGES_REQUESTED** [review]
Thanks for the awesome addition! I really appreciate the work. I have a couple minor suggested tweaks, if you don't mind?

File: src/unity.c (hunk 1 above)
Comment (Skinner927) [reply src/unity.c:2336]: Updated the options. If you'd prefer to choose exactly what to say I won't be offended.

P.S. TY for the fast review.

File: test/tests/test_generate_test_runner.rb (hunk 3 above)
Comment (Skinner927) [reply test/tests/test_generate_test_runner.rb:1166]: Marked as depreciated. I can remove it if you'd prefer.

**Pull request merged**